### PR TITLE
VSCode IDE

### PR DESCRIPTION
**Pull request by melloware**
@tandraschko @Rapster 

@FlipWarthog and myself have been using VSCode for all my PF development.  I like it because its so fast and easy.  However there is one annoying bug where Maven builds like `mvn clean jetty:run` performed outside of VSCode cause VSCode to start running its compile and the collide.

You can read the issue here but this fix seems to be the only one that makes both VSCode and command line play nicely: https://github.com/redhat-developer/vscode-java/issues/1381#issuecomment-846578339

This change should affect ABSOLUTELY nothing except anyone building PrimeFaces in either VSCode or Eclipse.   

> [!IMPORTANT]  
> IntelliJ and NetBeans are completely unaffected by this change.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -307,6 +309,15 @@
                 </plugins>
             </build>
         </profile>
+        <profile>
+            <id>vscode-ide</id>
+            <activation>
+                <property><name>m2e.version</name></property>
```
Comment (tandraschko) [inline pom.xml:315]: i wonder how is this related to vscode? i fear it could be activated in other cases, too

Comment (melloware) [reply pom.xml:315]: So VSCode and Eclipse both use the same M2M Eclipse plugin and that property is how its detected while the IDE is running.
